### PR TITLE
Rewrite of default password testing guide

### PR DESCRIPTION
**Pull request by rbsec**
As per discussion in #564, this guide had a lot of unnecessary fluff and padding.

### CODE REVIEW

**Comment by rbsec** [comment]
That link is not broken - seems to be a bug with the link checker.

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 1]
```diff
@@ -6,81 +6,68 @@
...
 
-When facing applications where we do not have a list of default and common user accounts (for example due to the fact that the application is not wide spread) we can attempt to guess valid default credentials. Note that the application being tested may have an account lockout policy enabled, and multiple password guess attempts with a known username may cause the account to be locked. If it is possible to lock the administrator account, it may be troublesome for the system administrator to reset it.
+Once the software has been identified, try to find whether it uses default passwords, and if so, what they are. This should include:
 
-Many applications have verbose error messages that inform the site users as to the validity of entered usernames. This information will be helpful when testing for default or guessable user accounts. Such functionality can be found, for example, on the log in page, password reset and forgotten password page, and sign up page. Once you have found a default username you could also start guessing passwords for this account.
+- Searching for "SOFTWARE default password".
```
Comment (ThunderSon) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:26]: Maybe just adding the brackets would help better? Just a suggestion
```suggestion
- Searching for "[SOFTWARE] default password".
```

**Review by ThunderSon — APPROVED** [review]

**Comment by rbsec** [comment]
@ThunderSon and @kingthorin - do there any further changes needed for this PR?

**Comment by kingthorin** [comment]
Sorry this one has been waiting on my review, it's been in my GitHub inbox for weeks. I'll try to knock it off this week.

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 2]
```diff
@@ -6,81 +6,68 @@
...
+Once the software has been identified, try to find whether it uses default passwords, and if so, what they are. This should include:
 
-Many applications have verbose error messages that inform the site users as to the validity of entered usernames. This information will be helpful when testing for default or guessable user accounts. Such functionality can be found, for example, on the log in page, password reset and forgotten password page, and sign up page. Once you have found a default username you could also start guessing passwords for this account.
+- Searching for "[SOFTWARE] default password".
+- Reviewing the manual or vendor documentation.
+- Checking common default password databases, such as [CIRT.net](https://cirt.net/passwords) or [SecLists Default Passwords](https://github.com/danielmiessler/SecLists/tree/master/Passwords/Default-Credentials).
+- Inspecting the application source code (if available).
+- Installing the application of a virtual machine and inspecting it.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:30]:
```suggestion
- Installing the application on a virtual machine and inspecting it.
```

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 3]
```diff
@@ -6,81 +6,68 @@
...
+- Inspecting the physical hardware for stickers (often present on network devices).
 
-More information about this procedure can be found in the following sections:
+If a default password can't be found, try common options such as:
 
-- [Testing for User Enumeration and Guessable User Account](../03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.md)
-- [Testing for Weak password policy](07-Testing_for_Weak_Password_Policy.md).
+- "admin", "password", "12345" or other [common default passwords](https://github.com/nixawk/fuzzdb/blob/master/bruteforce/passwds/default_devices_users%2Bpasswords.txt).
+- An empty or blank password.
+- The serial number of MAC address of the device.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:37]:
```suggestion
- The serial number of MAC address or the device.
```

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 4]
```diff
@@ -6,81 +6,68 @@
...
 
-Since these types of default credentials are often bound to administrative accounts you can proceed in this manner:
+If the username is unknown, there are various options for enumerating users, discussed in the [Testing for Account Enumeration](../03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.md) guide. Alternatively, try common options such as "admin", "root" or "system".
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:39]:
```suggestion
If the username is unknown, there are various options for enumerating users, discussed in the [Testing for Account Enumeration](../03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.md) guide. Alternatively, try common options such as "admin", "root", or "system".
```

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 5]
```diff
@@ -6,81 +6,68 @@
...
+- Inspecting the application source code (if available).
+- Installing the application of a virtual machine and inspecting it.
+- Inspecting the physical hardware for stickers (often present on network devices).
 
-More information about this procedure can be found in the following sections:
+If a default password can't be found, try common options such as:
 
-- [Testing for User Enumeration and Guessable User Account](../03-Identity_Management_Testing/04-Testing_for_Account_Enumeration_and_Guessable_User_Account.md)
-- [Testing for Weak password policy](07-Testing_for_Weak_Password_Policy.md).
+- "admin", "password", "12345" or other [common default passwords](https://github.com/nixawk/fuzzdb/blob/master/bruteforce/passwds/default_devices_users%2Bpasswords.txt).
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:35]:
```suggestion
- "admin", "password", "12345", or other [common default passwords](https://github.com/nixawk/fuzzdb/blob/master/bruteforce/passwds/default_devices_users%2Bpasswords.txt).
```

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md [hunk 6]
```diff
@@ -6,81 +6,68 @@
...
+When staff within an organisation manually create passwords for new accounts, they may do so in a predictable way. This can often be:
 
-It can also occur that when a new account is created in an application the account is assigned a default password. This password could have some standard characteristics making it predictable. If the user does not change it on first usage (this often happens if the user is not forced to change it) or if the user has not yet logged on to the application, this can lead an attacker to gain unauthorized access to the application.
+- A single common password such as "Password1".
+- Organisation specific details, such as the organisation name or address.
+- Passwords that follow a simple pattern, such as "Monday123" if account is created on a Monday.
 
-The advice given before about a possible lockout policy and verbose error messages are also applicable here when testing for default passwords.
+These type so of passwords are often difficult to identify from a black-box perspective, unless they can successfully be guessed or brute-forced. However, they are easy to identify when performing grey-box or white-box testing.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md:49]:
```suggestion
These types of passwords are often difficult to identify from a black-box perspective, unless they can successfully be guessed or brute-forced. However, they are easy to identify when performing grey-box or white-box testing.
```

**Review by kingthorin — CHANGES_REQUESTED** [review]
Just a few minor typos

File: document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md (hunk 3 above)
Comment (rbsec) [reply document/4-Web_Application_Security_Testing/04-Authentication_Testing/02-Testing_for_Default_Credentials.md]: This was fixed in 5277707 - the commit above changes the wrong "of".

**Comment by rbsec** [comment]
@kingthorin I've address all your comments.

**Review by kingthorin — APPROVED** [review]

**Pull request merged**